tests/loggers/test_base.py

```diff
@@ -225,31 +225,6 @@ def validation_epoch_end(self, outputs):
     trainer.fit(model)
 
 
-def test_with_accumulate_grad_batches():
-    """Checks if the logging is performed once for `accumulate_grad_batches` steps."""
-
-    class StoreHistoryLogger(CustomLogger):
-        def __init__(self):
-            super().__init__()
-            self.history = {}
-
-        @rank_zero_only
-        def log_metrics(self, metrics, step):
-            if step not in self.history:
-                self.history[step] = {}
-            self.history[step].update(metrics)
-
-    logger = StoreHistoryLogger()
-
-    np.random.seed(42)
-    for i, loss in enumerate(np.random.random(10)):
-        logger.agg_and_log_metrics({"loss": loss}, step=int(i / 5))
-
-    assert logger.history == {0: {"loss": 0.5623850983416314}}
-    logger.save()
-    assert logger.history == {0: {"loss": 0.5623850983416314}, 1: {"loss": 0.4778883735637184}}
-
-
```

Comment:
Since the aggregation logic is in the training loop, I'm not sure this test makes sense anymore

Reply:
We could always keep this test but just add a `pytest.deprecated_call` like here: https://github.com/PyTorchLightning/pytorch-lightning/blob/1b107c58928e75b42155aef4a3695b5ee243c276/tests/deprecated_api/test_remove_1-8.py#L46

Reply:
Currently the deprecation message is only shown at initialization if someone overrides agg_and_log_metrics in a custom logger. If we put a deprecation message in this method, it would spam it over and over each time it's called. Is that a behavior we're ok with?

Reply:
Hmm, yeah I see the problem, the way it is now we need a Trainer to emit the dep message. I'm not suggesting we put the dep message in `agg_and_log_metrics`